### ISSUE TITLE
Streaming overrides Cache-Control header to `no-cache`

### ISSUE THREAD

**Issue by aglushkov** [rails/rails#35312]
I stream files to CloudFront CDN using **ActionController::Live**,  and CDN caches all headers and sends them to browsers. And browsers never caches this files. That because rails sends `Cache-Control: no-cache` header, and CDN proxies it to browsers.

I wanted to set `Cache-Control` manually, but rails  [overrides](https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/live.rb#L149) this header silently.

I have a walk-around for now, but it is quite odd. I'm setting custom headers and then call `response.commit!`, so Rails thinks first chunk is already sent and does not overrides headers.

```ruby
response.delete_header 'Content-Length'
response.set_header 'Cache-Control', "max-age=#{1.year}, public"
response.commit!
```

### Steps to reproduce
```ruby
begin
  require "bundler/inline"
rescue LoadError => e
  $stderr.puts "Bundler version 1.10 or later is required. Please update your Bundler"
  raise e
end

gemfile(true) do
  source "https://rubygems.org"

  gem "rails", '~> 6.0.0.beta1'
end

require "rack/test"
require "action_controller/railtie"
require "minitest/autorun"

class TestApp < Rails::Application
  config.logger = Logger.new($stdout)
  Rails.logger  = config.logger
  config.hosts << 'www.example.com'

  routes.draw do
    get "/" => "test#index"
  end
end

class TestController < ActionController::Base
  include ActionController::Live

  def index
    response.set_header 'Cache-Control', "max-age=#{1.year}, public"
    response.stream.write('...')
  ensure
    response.stream.close
  end
end

class ExampleTest < ActionDispatch::IntegrationTest
  def test_login
    get "/"

    assert_equal "max-age=#{1.year}, public", headers['Cache-Control']
  end

  private

  def app
    Rails.application
  end
end

```
### Expected behavior
Headers provided by user should be sent without changes.
Rails should not override headers.
Cache-Control header `no-cache`  value could be default if nothing else provided. 
Maybe default should be no `Cache-Control` header at all.

### Actual behavior
Rails quietly overrides `Cache-Control` header to 'no-cache'.

### System configuration
**Rails version**: 5.2.2, 6.0.0.beta1
**Ruby version**: 2.5.0

**Comment by byellokore** [comment]
ActionController::Streaming does not override. Maybe, something like that could be implemented inside the ActionController::Live.

[https://github.com/rails/rails/blob/9c9c950d02af83742a5f76302d0faa99508f242c/actionpack/lib/action_controller/metal/streaming.rb#L207](url)

**Comment by kamipo** [comment]
Fixed by #35400.
